dipy/align/tests/test_mattes.py

```diff
@@ -0,0 +1,803 @@
+        theta = transform.get_identity_parameters()
+
+        static, moving, static_aff, moving_aff, smask, mmask, T = \
+            setup_random_transform(transform, factor, nslices, 5.0)
+        metric = MattesBase(32)
+        metric.setup(static, moving, smask, mmask)
+
+        # Compute the gradient at theta with the implementation under test
+        T = transform.param_to_matrix(theta)
+        shape = np.array(static.shape, dtype=np.int32)
+
+        warped = warp_method(moving.astype(np.float32), shape, T)
+        warped = np.array(warped)
+        metric.update_pdfs_dense(static.astype(np.float64),
+                                 warped.astype(np.float64))
+        J0 = np.copy(metric.joint)
+        # Now we have the joint distribution evaluated at theta
+        grid_to_space = np.eye(dim + 1)
+        spacing = np.ones(dim, dtype=np.float64)
+        mgrad, inside = vf.gradient(moving.astype(np.float32), moving_aff,
+                                    spacing, shape, grid_to_space)
+        id = transform.get_identity_parameters()
+        metric.update_gradient_dense(id, transform, static.astype(np.float64),
+                                     warped.astype(np.float64), grid_to_space,
+                                     mgrad, smask, mmask)
+        actual = np.copy(metric.joint_grad)
+        # Now we have the gradient of the joint distribution w.r.t. the
+        # transform parameters
+
+        # Compute the gradient using finite-diferences
+        n = transform.get_number_of_parameters()
+        expected = np.empty_like(actual)
+        for i in range(n):
+            dtheta = theta.copy()
+            dtheta[i] += h
+            # Update the joint distribution with the warped moving image
+            T = transform.param_to_matrix(dtheta)
+            shape = np.array(static.shape, dtype=np.int32)
+            warped = warp_method(moving.astype(np.float32), shape, T)
+            warped = np.array(warped)
+            metric.update_pdfs_dense(static.astype(np.float64),
+                                     warped.astype(np.float64))
+            J1 = np.copy(metric.joint)
+            expected[..., i] = (J1 - J0) / h
+
+        # Dot product and norms of gradients of each joint histogram cell
+        # i.e. the derivatives of each cell w.r.t. all parameters
+        P = (expected * actual).sum(2)
+        enorms = np.sqrt((expected ** 2).sum(2))
+        anorms = np.sqrt((actual ** 2).sum(2))
+        prodnorms = enorms * anorms
+        # Cosine of angle between the expected and actual gradients.
+        # Exclude very small gradients
+        P[prodnorms > 1e-6] /= (prodnorms[prodnorms > 1e-6])
+        P[prodnorms <= 1e-6] = 0
+        # Verify that a large proportion of the gradients point almost in
+        # the same direction. Disregard very small gradients
+        mean_cosine = P[P != 0].mean()
+        std_cosine = P[P != 0].std()
+        assert_equal(mean_cosine > 0.9, True)
```

Comment:
I couldn't find that assert here:
http://docs.scipy.org/doc/numpy/reference/routines.testing.html
`ImportError: cannot import name assert_true`
I tried simply "assert" and it worked, though =)